submissions/UA-girl/task_js_oop_exersise/index.js

```diff
@@ -0,0 +1,47 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/UA-girl/a-tiny-JS-world
+   Web app: https://ua-girl.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+
+class Creature {
+    constructor(species, name, age, gender, legs, say, friends = []) {
+        this.species = species;
+        this.name = name;
+        this.age = age;
+        this.gender = gender;
+        this.say = say;
+        this.legs = legs
+        this.friends = friends.length > 0 ? friends : [];
+    }
+
+    toString() {
+        return `${this.say}! My name is ${this.name} and I am ${this.age} years old ${this.gender} ${this.species}.  I have ${this.legs} legs and ${this.hasOwnProperty('hands') ? this.hands : 'no'} hands. My friends are ${this.friends.join(', ')}.`
```

Comment:
> There must be good reasons to handle properties outside the class that owns them

Class `Creature` doesn't own property `hands`. A base class should never be aware of child classes props. Let child classes handle their own props.

Imagine we enrich our world with other species, e.g. birds. Following current approach will make the base class unnecessary overwhelmed with "foreign" properties handling.

Employ method overriding where necessary.

A child class method may call base class method by means of `super.methodName()`

This will make your code cleaner.